### PR TITLE
[LOOP-5153] Remove HealthKit dependency from LoopAlgorithm

### PR DESCRIPTION
**Pull request by Camji55**
https://tidepool.atlassian.net/browse/LOOP-5153

### CODE REVIEW

File: Common/Models/WatchContext.swift [hunk 1]
```diff
@@ -63,11 +62,11 @@ final class WatchContext: RawRepresentable {
         isClosedLoop = rawValue["cl"] as? Bool
 
         if let unitString = rawValue["gu"] as? String {
-            displayGlucoseUnit = HKUnit(from: unitString)
+            displayGlucoseUnit = LoopUnit(from: unitString)
```
Comment (Camji55) [inline Common/Models/WatchContext.swift:65]: I will verify this -- not 100% this will work with mmol/L

File: Common/Models/WatchContext.swift [hunk 2]
```diff
@@ -126,7 +125,15 @@ final class WatchContext: RawRepresentable {
         raw["gc"] = glucoseCondition?.rawValue
         raw["gt"] = glucoseTrend?.rawValue
         if let glucoseTrendRate = glucoseTrendRate {
-            let unitPerMinute = unit.unitDivided(by: .minute())
+            let unitPerMinute: LoopUnit
```
Comment (Camji55) [inline Common/Models/WatchContext.swift:128]: Verify if needed

File: Loop Widget Extension/Timeline/StatusWidgetTimelineProvider.swift [hunk 3]
```diff
@@ -111,11 +111,21 @@ class StatusWidgetTimelineProvider: TimelineProvider {
             }
 
             let finalGlucose = glucose
+            
+            let unit: LoopUnit?
+            switch await healthStore.cachedPreferredUnits(for: .bloodGlucose) {
```
Comment (Camji55) [inline Loop Widget Extension/Timeline/StatusWidgetTimelineProvider.swift:116]: I need to verify if needed

File: Loop/Managers/LoopAppManager.swift [hunk 4]
```diff
@@ -298,7 +298,17 @@ class LoopAppManager: NSObject {
             }
 
             Task { @MainActor in
-                if let unit = await self.healthStore.cachedPreferredUnits(for: .bloodGlucose) {
+                var unit: LoopUnit?
+                switch await self.healthStore.cachedPreferredUnits(for: .bloodGlucose) {
```
Comment (Camji55) [inline Loop/Managers/LoopAppManager.swift:302]: I need to verify if needed

**Review by nhamming — APPROVED** [review]
LGTM

**Pull request merged**